lib/livereload.js

```diff
@@ -6,126 +6,116 @@
  *******************************************************
  */
 
-
-var path = require('path');
 var Q = require('q');
-var fs = require('fs');
-var url = require('url');
-var logger = require('./utils/logger');
+var path = require('path');
+var multiPlatforms = require('./utils/platforms');
 var helpers = require('./utils/helpers');
-var merge = require('lodash.merge');
+var Patcher = require('./utils/Patcher');
+var livereloadEvents = require('./utils/events');
 
-var projectRoot,
-    platformWWWDirs;
+/** This file is exposed externally */
 
-module.exports = LiveReload;
 
-/**
- * @constructor
- * @param {string} projectRoot - root folder of the project
- * @param {string[]} platformWWWs - www directories of the platforms
- * @param {Object} options - Options to initialize LiveReload with - These are to be passed to the underlying BrowserSync.
- *                           See http://www.browsersync.io/docs/options/ for documentation
+/** Start the BrowserSync server and start livereloading
+ *  @param {string} projectRoot - The root of the cordova project being dealt with.
+ *  @param {Array.<string[]>} platforms - The platforms we want to livereload.
+ *  @param options - Options
+ *  @param options.ignore - files to ignore.
+ *  @param options.tunnel - use the --tunnel option to get around firewall/proxy issues.
+ *  @param options.ghostMode - sync gestures (form inputs, clicks, scrolls) across multiple devices.
+ *  @param options.cb - callback function to be called when .
  */
-function LiveReload(projectRoot, platformWWWs, options) {
-    this.projectRoot = projectRoot;
-    this.startPage = helpers.GetStartPage(this.projectRoot);
-    this.www_dir = path.join(this.projectRoot, 'www');
-    platformWWWDirs = platformWWWs;
-    this.startServer = startServer;
-    this.stopServer = stopServer;
-    this.options = options;
-    this.browserSync = require('browser-sync').create(); // If this is exposed to users, how would they handle deps install ?
-    return this;
-};
-
-
-function startServer() {
-    var self = this;
-    var deferred = Q.defer();
+module.exports.start = function (projectRoot, platforms, options) {
+    
+    // Validate whether all platforms are currently supported
+    // If a platform is not supported by the plugin, display an error message and don't process it any further
+    platforms.forEach(function (plat, index, platforms) {
+        if (!multiPlatforms.isPlatformSupported(plat)) {
+            var msg = 'The "' + plat + '" platform is not supported.';
+            livereloadEvents.emit('livereload:warning', msg);
+            platforms.splice(index, 1);
+        }
+    });
 
-    var defaultOptions = {
-        server: {
-            baseDir: self.projectRoot,
-            directory: true
-        },
-        open: false,
+    // If none of the supplied platforms are supported, stop and return an error
+    if (platforms === undefined || platforms.length === 0) {
```

Comment:
I think the platforms===undefined check needs to be done before the forEach above.